avocado_vt/test.py

```diff
@@ -217,28 +218,48 @@ def setUp(self):
         Avocado-vt uses custom setUp/test/tearDown handling and unlike
         Avocado it allows skipping tests from any phase. To convince
         Avocado to allow skips let's say our tests run during setUp
-        phase and don't do anything during runTest/tearDown.
+        phase and report the status in test.
         """
         env_lang = os.environ.get('LANG')
         os.environ['LANG'] = 'C'
         try:
             self._runTest()
+            self.__status = "PASS"
         # This trick will give better reporting of virt tests being executed
         # into avocado (skips, warns and errors will display correctly)
-        except error.TestNAError, details:
-            raise exceptions.TestSkipError(details)
-        except error.TestWarn, details:
-            raise exceptions.TestWarn(details)
-        except error.TestError, details:
-            raise exceptions.TestError(details)
-        except error.TestFail, details:
-            raise exceptions.TestFail(details)
+        except:  # Old-style exceptions are not inherited from Exception()
+            details = sys.exc_info()[1]
+            self.__status = details
+            if not hasattr(self, "cancel"):     # Old Avocado, skip here
+                if isinstance(self.__status, error.TestNAError):
+                    if hasattr(exceptions, "TestSetupSkip"):
+                        raise exceptions.TestSetupSkip(self.__status)
```

Comment:
True, I'm sorry I mistaken the latest changes to Skip exceptions in Avocado. I'll double check and update the PR.